waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -131,3 +132,14 @@ backwardsCompatibleWith = ComparatorSet . pure . BackwardsCompatibleWith
 
 mkPrimCompSet :: Operator -> Version -> ComparatorSet
 mkPrimCompSet op = ComparatorSet . pure . PrimitiveComparator op
+
+doesVersionRangeAllowMajorChanges :: Range -> Bool
+doesVersionRangeAllowMajorChanges = not . doesVersionRangeAllowOnlyMinorChanges
+  where
+    doesVersionRangeAllowOnlyMinorChanges versionRange = isJust $ do
```

Comment:
This is a local function but seems to be the one we actually need in `Valid.hs`.

Reply:
I didn't get this comment, what do you mean? We use `doesVersionRangeAllowMajorChanges` in Valid.hs, which is this function that uses this other function in `where`.

Reply:
We do (in pseudocode):
```
// definition
doesVersionRangeAllowMajorChanges = not doesVersionRangeAllowOnlyMinorChanges
  where doesVersionRangeAllowOnlyMinorChanges = ...

// caller
import doesVersionRangeAllowMajorChanges
checkUserRangeAllowsOnlyMinorChanges = if doesVersionRangeAllowMajorChanges, return error
```
Why not:
```
// definition
doesVersionRangeAllowOnlyMinorChanges = ...

// caller
import doesVersionRangeAllowOnlyMinorChanges
checkUserRangeAllowsOnlyMinorChanges = if not doesVersionRangeAllowOnlyMinorChanges, return error
```
Less back and forth and less knowledge needed to verify correctness, so I was just curious why we don't use the condition we need directly.

Either way, no big deal, it can stay. You can resolve the comment no matter what you go for.

Reply:
Ok got it. Yeah, I did this on purpose, because I believe it is easier to read `if doesVersionRangeAllowMajorChanges` in Valid.hs, then to read `if not doesVersionRangeAllowOnlyMinorChanges`. For the later, you have to reason about it quite a bit to figure it really means that only major changes are allowed (at least I had to).

On the other hand, it was easier for me to implement `doesVersionRangeAllowOnlyMinorChanges` when implementing it. 

So I combined the two: what is easier to implement, and what I actually need at the call site, and ended up with this, which is the best.

Ok, leaving this then!

Reply:
Ok, now got what you mean from live discussion -> fixed!